garden_ai/app/model.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+from typing import Optional, List
+
+from garden_ai.client import GardenClient
+
+import typer
+import rich
+
+model_app = typer.Typer(name="model", no_args_is_help=True)
+
+
+@model_app.callback()
+def pipeline():
+    """
+    sub-commands for managing machine learning models
+    """
+    pass
+
+
+@model_app.command(no_args_is_help=True)
+def register(
+    name: str = typer.Argument(
+        None,
+        help=("The name of your model"),
+    ),
+    model_path: Path = typer.Argument(
+        None,
+        dir_okay=False,
+        file_okay=True,
+        writable=True,
+        readable=True,
+        resolve_path=True,
+        help=("The path to your model on your filesystem"),
+    ),
+    flavor: str = typer.Argument(
+        "sklearn",
+        help=(
+            "What ML library did you make the model with? "
+            "Currently we just support sklearn."
+        ),
+    ),
+    extra_pip_requirements: Optional[List[str]] = typer.Option(
+        None,
```

Comment:
we should probably add a short name like -x (just `Option(None, "--extra_pip_requirements", "-x", help="blah" )` instead